pkg/queue/readiness/probe.go

```diff
@@ -128,36 +148,40 @@ func (p *Probe) ProbeContainer() bool {
 }
 
 func (p *Probe) probeContainerImpl() bool {
-	var err error
-
-	switch {
-	case p.HTTPGet != nil:
-		err = p.httpProbe()
-	case p.TCPSocket != nil:
-		err = p.tcpProbe()
-	case p.GRPC != nil:
-		err = p.grpcProbe()
-	case p.Exec != nil:
-		// Should never be reachable. Exec probes to be translated to
-		// TCP probes when container is built.
-		// Using Fprintf for a concise error message in the event log.
-		fmt.Fprintln(p.out, "exec probe not supported")
-		return false
-	default:
-		// Using Fprintf for a concise error message in the event log.
-		fmt.Fprintln(p.out, "no probe found")
-		return false
+	var probeGroup errgroup.Group
+
+	for _, probe := range p.probes {
+		innerProbe := probe
+		probeGroup.Go(func() error {
+			fmt.Println("Probing", innerProbe)
+			switch {
+			case innerProbe.HTTPGet != nil:
+				return innerProbe.httpProbe()
+			case innerProbe.TCPSocket != nil:
+
+				return innerProbe.tcpProbe()
+			case innerProbe.GRPC != nil:
+				return innerProbe.grpcProbe()
+			case innerProbe.Exec != nil:
+				// Should never be reachable. Exec probes to be translated to
+				// TCP probes when container is built.
+				return fmt.Errorf("exec probe not supported")
+			default:
+				return fmt.Errorf("no probe found")
+			}
+		})
 	}
 
+	err := probeGroup.Wait()
```

Comment:
we aggregate the results of all readiness-probes to one result on QP

Reply:
Could we use `sync.WaitGroup` so we collect all errors instead here?